vm/src/obj/objstr.rs

```diff
@@ -429,6 +430,29 @@ impl PyString {
         }
     }
 
+    #[pymethod(name = "__mod__")]
+    fn modulo(&self, values: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        let format_string_text = &self.value;
+
+        match CFormatString::from_str(format_string_text.as_str()) {
+            Ok(format_string) => perform_clike_format(vm, format_string, values.clone()),
+            Err(err) => match err.typ {
+                CFormatErrorType::UnsupportedFormatChar(c) => Err(vm.new_value_error(format!(
```

Comment:
Instead of rendering the `CFormatErrorType` to a string, implement the `Display` trait for `CFormatErrorType`, or add a method to `CFormatErrorType` called `to_string` or something like that. Then this code can resolve to:
```rust
fn module() {

    let format_string = CFormatString::from_str(format_string_text)
             .map_err(|err| vm.new_value_error(err.to_string()))?;

    perform_clike_format(vm, format_string_text)

}
```

Reply:
Thanks for the good advice!
I had in mind though that the cformat crate will only do parsing and formatting, and won't have to do with any python internals by itself.

Reply:
You're right that the parsing of format strings does not require any python objects. Are you going to implement the change I suggested above?

Reply:
Yeah, I've pushed it